src/cdk-experimental/table-scroll-container/table-scroll-container.ts

```diff
@@ -0,0 +1,136 @@
+    OnDestroy {
+  protected readonly CLASS_NAME = 'cdk-table-scroll-container';
+
+  private readonly _document: Document;
+  private readonly _element: HTMLElement;
+  private readonly _uniqueClassName: string;
+  private readonly _ruleIndexes = new Map<Orientation, number>();
+
+  private _styleElement?: HTMLStyleElement;
+  private _indexSequence = 0;
+  private _startSizes: (number|null|undefined)[] = [];
+  private _endSizes: (number|null|undefined)[] = [];
+  private _headerSizes: (number|null|undefined)[] = [];
+  private _footerSizes: (number|null|undefined)[] = [];
+
+  constructor(
+      @Optional() private readonly _directionality: Directionality,
+      elementRef: ElementRef<HTMLElement>,
+      @Inject(DOCUMENT) document: any) {
+    this._document = document;
+    this._element = elementRef.nativeElement;
+    this._uniqueClassName = `cdk-table-scroll-container-${++nextId}`;
+    this._element.classList.add(this.CLASS_NAME);
+    this._element.classList.add(this._uniqueClassName);
+  }
+
+  ngOnDestroy(): void {
+    // TODO: Use remove() once we're off IE11.
+    if (this._styleElement?.parentNode) {
+      this._styleElement.parentNode.removeChild(this._styleElement);
+      this._styleElement = undefined;
+    }
+  }
+
+  stickyColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._startSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyEndColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._endSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyHeaderRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._headerSizes = sizes;
+    this._updateVertical();
+  }
+
+  stickyFooterRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._footerSizes = sizes;
+    this._updateVertical();
+  }
+
+  private _updateHorizontal(): void {
+    const startMargin = computeMargin(this._startSizes);
+    const endMargin = computeMargin(this._endSizes);
+
+    if (startMargin === 0 && endMargin === 0) {
+      this._applyCss('horizontal', '');
+      return;
+    }
+
+    const direction = this._directionality ? this._directionality.value : 'ltr';
+    const leftMargin = direction === 'rtl' ? endMargin : startMargin;
+    const rightMargin = direction === 'rtl' ? startMargin : endMargin;
+
+    this._applyCss('horizontal', `0 ${rightMargin}px 0 ${leftMargin}px`);
+  }
+
+  private _updateVertical(): void {
+    const topMargin = computeMargin(this._headerSizes);
+    const bottomMargin = computeMargin(this._footerSizes);
+
+    if (topMargin === 0 && bottomMargin === 0) {
+      this._applyCss('vertical', '');
+      return;
+    }
+
+    this._applyCss('vertical', `${topMargin}px 0 ${bottomMargin}px`);
+  }
+
+  private _getStyleSheet(): CSSStyleSheet {
+    if (!this._styleElement) {
+      this._styleElement = this._document.createElement('style');
+      this._styleElement.appendChild(this._document.createTextNode(''));
```

Comment:
I don't think you need the extra text node here since you're inserting styles through `insertRule`.